libcontainer/utils/utils_unix.go

```diff
@@ -98,3 +107,74 @@ func NewSockPair(name string) (parent *os.File, child *os.File, err error) {
 	}
 	return os.NewFile(uintptr(fds[1]), name+"-p"), os.NewFile(uintptr(fds[0]), name+"-c"), nil
 }
+
+type schedAttr struct {
+	Size          uint32
+	SchedPolicy   uint32
+	SchedFlags    uint64
+	SchedNice     int32
+	SchedPriority uint32
+	SchedRuntime  uint64
+	SchedDeadline uint64
+	SchedPeriod   uint64
+}
+
+// SetSchedAttr sets the scheduler attributes for the process with the given pid.
+// Please refer to the following link for kernel-specific values:
+// https://github.com/torvalds/linux/blob/c1a515d3c0270628df8ae5f5118ba859b85464a2/include/uapi/linux/sched.h#L111-L134
+func SetSchedAttr(pid int, scheduler *configs.Scheduler) error {
```

Comment:
Is there a reason `golang.org/x/sys/unix` doesn't have a wrapper for this? This syscall is fairly old -- first added in Linux 3.14.

Reply:
My best guess is "no one bothered to". Let me give it a try.

Reply:
Here's my attempt https://go-review.googlesource.com/c/sys/+/516756

(my best guess is "because of this ugly linux/glibc header conflict" now)

Reply:
@utam0k the above PR is now merged; can you please vendor latest (@HEAD) x/sys/unix and use `unix.SetSchedAttr`?

Alas, this PR will have to be marked as draft until x/sys/unix makes a release.

Reply:
https://github.com/opencontainers/runc/pull/4006 comes with SchedSetAttr